src/libprojectM/Renderer/BeatDetect.hpp

```diff
@@ -40,18 +40,18 @@ class BeatDetect
 {
 public:
     // Does this really belong here? maybe belongs on projectM.Settings?
-    float beatSensitivity;
+    float beatSensitivity = 1.f;
```

Comment:
I can change this if this is the preferred style. What is a situation where unnecessary copy constructors would be called with assignment?

Reply:
If you initialize class types:
```c++
// Calls the constructor, then the copy or move assignment operator
MyClass inst = MyClass(someValue);
// Will create the instance in-place, only calling the constructor
MyClass inst{ someValue };
// Same as old-style parenthesis syntax:
MyClass inst(someValue);
```

Otherwise, it's just for readability, seeing the `{}` immediately tells you it's an initialization.

Reply:
Hmm, I thiiink the first one will elide the copy, and that this behaviour is enforced by the standard?

Reply:
https://en.cppreference.com/w/cpp/language/copy_elision

Reply:
@kblaschke copy elision also is what makes [AAA](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) style viable.